From e9e06d9cd0b97901c79f4f9490320fb270ed5725 Mon Sep 17 00:00:00 2001
From: Paul Gofman <pgofman@codeweavers.com>
Date: Thu, 30 Apr 2026 12:40:18 -0600
Subject: [PATCH 33/47] wininet: Pass proxy server addr_str if present in
 open_http_connection().

If proxy is present request->server->addr_str is empty string.

CW-Bug-Id: #27010
---
 dlls/wininet/http.c | 11 ++++++-----
 1 file changed, 6 insertions(+), 5 deletions(-)

diff --git a/dlls/wininet/http.c b/dlls/wininet/http.c
index c1bba8a6594..ab6f8894894 100644
--- a/dlls/wininet/http.c
+++ b/dlls/wininet/http.c
@@ -4888,6 +4888,7 @@ static void http_process_keep_alive(http_request_t *req)
 
 static DWORD open_http_connection(http_request_t *request, BOOL *reusing)
 {
+    server_t *server;
     netconn_t *netconn = NULL;
     DWORD res;
 
@@ -4936,13 +4937,13 @@ static DWORD open_http_connection(http_request_t *request, BOOL *reusing)
 
     TRACE("connecting to %s, proxy %s\n", debugstr_w(request->server->name),
           request->proxy ? debugstr_w(request->proxy->name) : "(null)");
-
+    server = request->proxy ? request->proxy : request->server;
     INTERNET_SendCallback(&request->hdr, request->hdr.dwContext,
                           INTERNET_STATUS_CONNECTING_TO_SERVER,
-                          request->server->addr_str,
-                          strlen(request->server->addr_str)+1);
+                          server->addr_str,
+                          strlen(server->addr_str)+1);
 
-    res = create_netconn(request->proxy ? request->proxy : request->server, request->security_flags,
+    res = create_netconn(server, request->security_flags,
                          (request->hdr.ErrorMask & INTERNET_ERROR_MASK_COMBINED_SEC_CERT) != 0,
                          request->hdr.connect_timeout, &netconn);
     if(res != ERROR_SUCCESS) {
@@ -4954,7 +4955,7 @@ static DWORD open_http_connection(http_request_t *request, BOOL *reusing)
 
     INTERNET_SendCallback(&request->hdr, request->hdr.dwContext,
             INTERNET_STATUS_CONNECTED_TO_SERVER,
-            request->server->addr_str, strlen(request->server->addr_str)+1);
+            server->addr_str, strlen(server->addr_str)+1);
 
     *reusing = FALSE;
     TRACE("Created connection to %s: %p\n", debugstr_w(request->server->name), netconn);
-- 
2.54.0

